libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -333,6 +333,23 @@ int WiFiClient::getSocketOption(int level, int option, const void* value, size_t
     return res;
 }
 
+int WiFiClient::setConnectionTimeout(uint32_t milliseconds)
+{
+    _timeout = milliseconds;
+    if(fd() >= 0) {
+        struct timeval tv;
+        tv.tv_sec = _timeout / 1000;
+        tv.tv_usec = (_timeout  % 1000) * 1000;
+        if(setSocketOption(SO_RCVTIMEO, (char *)&tv, sizeof(struct timeval)) < 0) {
+            return -1;
+        }
+        return setSocketOption(SO_SNDTIMEO, (char *)&tv, sizeof(struct timeval));
+    }
+    else {
+        return 0;
+    }
+}
+
```

Comment:
As the timeout will be set in the `write()` and `read()` functions anyways, because the `_lastReadTimeout` and `_lastWriteTimeout` remains unchanged, and also the Arduino standart says to be void function, I recommend changing the function just to set the `_timeout` variable like this:
```suggestion
void WiFiClient::setConnectionTimeout(uint32_t milliseconds)
{
    _timeout = milliseconds;
}

```